lib/ProxySQL_Statistics.cpp

```diff
@@ -1068,3 +1079,103 @@ void ProxySQL_Statistics::MySQL_Query_Cache_sets(SQLite3_result *resultset) {
 		statsdb_disk->execute(buf);
 	}
 }
+
+bool ProxySQL_Statistics::knows_variable_name(const std::string & variable_name) const
+{
+	return (variable_name_id_map.find(variable_name) != variable_name_id_map.end());
+}
+
+int64_t ProxySQL_Statistics::get_variable_id_for_name(const std::string & variable_name) {
+	lock_guard<mutex> lock(mu);
+
+	int64_t variable_id = -1; // Negative value indicates not yet found.
+
+	if (knows_variable_name(variable_name)) {
+		variable_id = variable_name_id_map[variable_name];
+	} else {
+		// No matching variable_id found in map. Try loading from the SQLite lookup table on disk 
+		SQLite3_result *result = NULL;
+		int cols;
+		int affected_rows;
+		char *error = NULL;
+		
+		auto select_var_id = [&]() -> int64_t {
+			const string var_id_query = "SELECT variable_id FROM history_mysql_status_variables_lookup WHERE variable_name=\"" + variable_name + "\"";
+			statsdb_disk->execute_statement(var_id_query.c_str(), &error , &cols , &affected_rows , &result);
+
+			if (error) {
+				proxy_error("SQLITE CRITICAL ERROR %s. Shutting down\n", error);
+				free(error);
+				error = NULL;
+				exit(EXIT_SUCCESS); 
+			} 
+			
+			if (result) {
+				if (result->rows_count > 0) {
+					// matching variable_id for variable_name in lookup table
+					SQLite3_row *r = result->rows[0];
+					return strtoll(r->fields[0], NULL, 10);
```

Comment:
There is a memory leak here.
`result` is not deleted before returning

Reply:
Good catch